inc/c_util/tarray_ll.h

```diff
@@ -99,6 +105,51 @@ TARRAY_LL(T) TARRAY_LL_CREATE(C)(void)
+            /* Codes_SRS_TARRAY_01_005: [ If there are any failures then TARRAY_CREATE_WITH_CAPACITY(T) shall fail and return NULL. ]*/ \
+            LogError("failure in " MU_TOSTRING(THANDLE_MALLOC) "(" MU_TOSTRING(TARRAY_TYPEDEF_NAME(T)) "=%zu)", sizeof(TARRAY_TYPEDEF_NAME(T))); \
+            /*return as is*/                                                                                                \
+        }                                                                                                                   \
+        else                                                                                                                \
+        {                                                                                                                   \
+            /* Codes_SRS_TARRAY_01_003: [ TARRAY_CREATE_WITH_CAPACITY(T) shall call malloc_2 to allocate capacity entries for result->arr. ]*/ \
+            result->arr = malloc_2(capacity,sizeof(T));                                                                     \
+            if(result->arr == NULL)                                                                                         \
+            {                                                                                                               \
+                /* Codes_SRS_TARRAY_01_005: [ If there are any failures then TARRAY_CREATE_WITH_CAPACITY(T) shall fail and return NULL. ]*/ \
+                LogError("failure in malloc(1 * sizeof(" MU_TOSTRING(T) ")=%zu)",                                           \
+                    sizeof(T));                                                                                             \
+            }                                                                                                               \
+            else                                                                                                            \
+            {                                                                                                               \
+                result->capacity = capacity;                                                                                \
+                /* Codes_SRS_TARRAY_01_004: [ TARRAY_CREATE_WITH_CAPACITY(T) shall succeed and return a non-NULL value. ]*/ \
+                goto allok;                                                                                                 \
+            }                                                                                                               \
+            THANDLE_FREE(TARRAY_TYPEDEF_NAME(C))(result);                                                                   \
+        }                                                                                                                   \
+        result = NULL;                                                                                                      \
```

Comment:
>result = NULL; [](http://example.com/codeflow?start=8&length=14)

if line 125 executes, then this line executes and sets "result" to NULL (again).

Move this line between 145, 146.

Reply:
So good to have tests for these small refactors.